subprojects/version-control/src/main/java/org/gradle/vcs/internal/VersionControlServices.java

```diff
@@ -37,8 +38,8 @@ VcsMappingFactory createVcsMappingFactory() {
             return new DefaultVcsMappingFactory();
         }
 
-        VersionControlSystemFactory createVersionControlSystemFactory() {
-            return new DefaultVersionControlSystemFactory();
+        VersionControlSystemFactory createVersionControlSystemFactory(CacheRepository cacheRepository) {
```

Comment:
We'll need to move this service into build session scope or lower, that's when `CacheRepository` is available.

Reply:
I'll happily do that. But, how did you know?

Reply:
When I tried creating the integ test, it exploded trying to create this service.

I looked at where this is being created (`VersionControlGlobalServices`) and its dependencies (`CacheRepository`). I then looked for an implementation of that dependency (`DefaultCacheRepository`) and where it's created (`CacheRepositoryServices`).  That class isn't a `AbstractPluginServiceRegistry` so it must be created somewhere else.  I see an instance created in `BuildSessionScopeServices` and `GradleUserHomeScopeServices`, which make a `CacheRepository` available in each of those scopes.

If you have questions about the ordering of the scopes, the methods in `PluginServiceRegistry` are in the order of the scopes.

Reply:
Makes sense. I've moved it.